Lib/shutil.py

```diff
@@ -511,8 +511,20 @@ def onerror(*args):
 rmtree.avoids_symlink_attacks = _use_fd_functions
 
 def _basename(path):
-    # A basename() variant which first strips the trailing slash, if present.
-    # Thus we always get the last component of the path, even for directories.
+    """A basename() variant which first strips the trailing slash, if present.
+    Thus we always get the last component of the path, even for directories.
+
+    path: Union[PathLike, str]
```

Comment:
Maybe it's just me, but this docstring seems unnecessarily verbose for such a simple function. The wording is good enough that it's pretty clear what's happening without needing the examples below.